src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonICollectionOfTConverter.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    /// <summary>
+    /// Converter for <cref>System.Collections.Generic.ICollection{TElement}</cref>.
+    /// </summary>
+    internal sealed class JsonICollectionOfTConverter<TCollection, TElement>
+        : JsonIEnumerableDefaultConverter<TCollection, TElement>
+        where TCollection : ICollection<TElement>
+    {
+        protected override void Add(TElement value, ref ReadStack state)
+        {
+            Debug.Assert(state.Current.ReturnValue is ICollection<TElement>);
+            ((ICollection<TElement>)state.Current.ReturnValue!).Add(value);
+        }
+
+        protected override void CreateCollection(ref ReadStack state, JsonSerializerOptions options)
+        {
+            JsonClassInfo classInfo = state.Current.JsonClassInfo;
+
+            if (TypeToConvert.IsInterface || TypeToConvert.IsAbstract)
+            {
+                if (!TypeToConvert.IsAssignableFrom(RuntimeType))
+                {
+                    ThrowHelper.ThrowNotSupportedException_DeserializeNoParameterlessConstructor(TypeToConvert);
+                }
+
+                state.Current.ReturnValue = new List<TElement>();
+            }
+            else
+            {
+                if (classInfo.CreateObject == null)
+                {
+                    ThrowHelper.ThrowNotSupportedException_DeserializeNoParameterlessConstructor(TypeToConvert);
+                }
+
+                TCollection returnValue = (TCollection)classInfo.CreateObject!()!;
```

Comment:
We already checked that `classInfo.CreateObject` can't be null above. Remove the `!`

```suggestion
                TCollection returnValue = (TCollection)classInfo.CreateObject()!;
```

Reply:
Sure. I'll fix instances of this. This is going to be a maintenance issue long-term unless we get a compiler warning.